Documentation/Books/Manual/Programs/Arangorestore/FastClusterRestore.md

```diff
@@ -0,0 +1,279 @@
+This step is **not** needed if you are using v3.3.22 or higher or v3.4.2 or higher
+and you have used in the previous step the option `--replication-factor 1`.
+{% endhint %}
+
+To speed up restore, it is possible to set the _replication factor_ to 1 before
+importing any data. Run the following command from exactly one _Coordinator_ (any
+_Coordinator_ can be used):
+
+```
+echo 'db._collections().filter(function(c) { return c.name()[0] !== "_"; })
+.forEach(function(c) { print("collection:", c.name(), "replicationFactor:",
+c.properties().replicationFactor); c.properties({ replicationFactor: 1 }); });'
+| arangosh
+  --server.endpoint <endpoint-of-a-coordinator>
+  --server.database <database-name>
+  --server.username <user-name>
+  --server.password <password>
+```
+
+Step 4: Create parallel restore scripts
+---------------------------------------
+
+Now that the Cluster is prepared, the `parallelRestore` script will be used.
+
+Please create the below `parallelRestore` script in any of your _Coordinators_.
+
+When executed (see below for further details), this script will create other scripts
+that can be then copied and executed on each _Coordinator_.
+
+
+```
+#!/bin/sh
+#
+# Version: 0.3
+#
+# Release Notes:
+# - v0.3: fixed a bug that was happening when the collection name included an underscore 
+# - v0.2: compatibility with version 3.4: now each coordinator_<number-of-coordinator>.sh
+#         includes a single restore command (instead of one for each collection)
+#         which allows making using of the --threads option in v.3.4.0 and later
+# - v0.1: initial version
+
+if test -z "$ARANGOSH" ; then
+  export ARANGOSH=arangosh
+fi
+cat > /tmp/parallelRestore$$.js <<'EOF'
+var fs = require("fs");
+var print = require("internal").print;
+var exit = require("internal").exit;
+var arangorestore = "arangorestore";
+var env = require("internal").env;
+if (env.hasOwnProperty("ARANGORESTORE")) {
+  arangorestore = env["ARANGORESTORE"];
+}
+
+// Check ARGUMENTS: dumpDir coordinator1 coordinator2 ...
+
+if (ARGUMENTS.length < 2) {
+  print("Need at least two arguments DUMPDIR and COORDINATOR_ENDPOINTS!");
+  exit(1);
+}
+
+var dumpDir = ARGUMENTS[0];
+var coordinators = ARGUMENTS[1].split(",");
+var otherArgs = ARGUMENTS.slice(2);
+
+// Quickly check the dump dir:
+var files = fs.list(dumpDir).filter(f => !fs.isDirectory(f));
+var found = files.indexOf("ENCRYPTION");
+if (found === -1) {
+  print("This directory does not have an ENCRYPTION entry.");
+  exit(2);
+}
+// Remove ENCRYPTION entry:
+files = files.slice(0, found).concat(files.slice(found+1));
+
+for (let i = 0; i < files.length; ++i) {
+  if (files[i].slice(-5) !== ".json") {
+    print("This directory has files which do not end in '.json'!");
+    exit(3);
+  }
+}
+files = files.map(function(f) {
+  var fullName = fs.join(dumpDir, f);
+  var collName = "";
+  if (f.slice(-10) === ".data.json") {
+    var pos;
+    if (f.slice(0, 1) === "_") {  // system collection

```

Comment:
This part of the script will not relyably work, i think i had discussion with @lservini on how to fix this,

Reply:
seems fixed per luigi. thanks for reviewing. improvements might be done in a new pr.